src/main/java/retriever/task/TaskList.java

```diff
@@ -0,0 +1,217 @@
+package retriever.task;
+
+import java.util.ArrayList;
+
+import retriever.Storage;
+import retriever.Ui;
+import retriever.exception.IllegalDateFormatException;
+import retriever.exception.IllegalDeadlineFormatException;
+import retriever.exception.IllegalEventFormatException;
+import retriever.exception.IllegalTaskNumberException;
+import retriever.exception.IllegalTodoFormatException;
+import retriever.exception.TaskNotFoundException;
+
+/**
+ * This class handles the task list and various
+ * operations on it.
+ */
+public class TaskList {
+    private ArrayList<Task> userTaskList;
+    private Storage taskStorage;
```

Comment:
I feel TaskList doesnt have a HAS-A relationship with Storage? So perhaps it would be better if Storage was in charge reading from the database and returning a ArrayList of tasks for TaskList to initialize